app/src/main/java/academy/apparchitects/notesapp/ui/screens/note_list/NotesListScreen.kt

```diff
@@ -0,0 +1,112 @@
+package academy.apparchitects.notesapp.ui.screens.note_list
+
+import academy.apparchitects.notesapp.data.Note
+import academy.apparchitects.notesapp.presentation.noteslist.NotesListStates
+import academy.apparchitects.notesapp.presentation.noteslist.NotesListVM
+import academy.apparchitects.notesapp.ui.components.Loader
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FabPosition
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun NotesListScreen(
+  onNoteClick: (String, Note) -> Unit,
+  onAddNoteClick: () -> Unit,
+  notesListVM: NotesListVM = hiltViewModel(),
+  modifier: Modifier = Modifier
+) {
+  val state = notesListVM.state.collectAsStateWithLifecycle()
+
+  LaunchedEffect(key1 = Unit) {
+    notesListVM.fetchNotes()
+  }
+
+  Scaffold(
```

Comment:
Put Scaffold and Topbar in root. No need to put scaffold and topbar repeatedly in every screen